lib/PreviewProvider.js

```diff
@@ -209,11 +210,29 @@ PreviewProvider.prototype = {
   }),
 
   /**
-   * Find the metadata for each link in the database
+   * Find the metadata for each link in the database. Note that it'll try to fetch items
+   * from the cache before querying the metadata store
    */
   _asyncFindItemsInDB: Task.async(function*(links) {
-    const cacheKeyArray = links.map(link => link.cache_key);
-    let linksMetadata = yield this._metadataStore.asyncGetMetadataByCacheKey(cacheKeyArray);
+    let cacheMissedKeys = [];
+    let linksMetadata = [];
+
+    // Hit the cache first
+    links.forEach(link => {
+      const key = link.cache_key;
+      const fromCache = MetadataCache.cache.get(key);
+      if (fromCache) {
+        linksMetadata.push(fromCache);
+      } else {
+        cacheMissedKeys.push(key);
```

Comment:
just to be sure, ```MetadataCache.cache.get(key)``` will return just the link's cache_key right? In other words you don't need to make a ```cacheKeyArray``` right?